pkg/api/v1beta1/types.go

```diff
@@ -412,6 +414,33 @@ type ContainerStatus struct {
 	ContainerID string `json:"containerID,omitempty" description:"container's ID in the format 'docker://<container_id>'"`
 }
 
+type PodConditionKind string
+
+// These are valid conditions of pod.
+const (
+	// PodReady means the pod is able to service requests and should be added to the
+	// load balancing pools of all matching services.
+	PodReady PodConditionKind = "Ready"
+)
+
+type PodConditionStatus string
```

Comment:
We should just update v1beta1 and v1beta2 to use ConditionStatus. As long as we don't change the string contents or field names, the type names shouldn't matter. They don't even get exposed via the swagger API at the moment.